### PR TITLE
Unit Parsing library

### PR DESCRIPTION
**Pull request by FelixBenning**
implement #31

### CODE REVIEW

**Comment by eike-fokken** [comment]
Looks good so far, but please make some more test cases, namely for these units (these all appear in the topology.json):
```
                    "unit": "1000m_cube_per_hour",
                    "unit": "bar",
                    "unit": "Celsius",
                    "unit": "K",
                    "unit": "km",
                    "unit": "kW",
                    "unit": "m",
                    "unit": "m_cube_per_s",
                    "unit": "mm",
                    "unit": "per_min",
```
I'm not sure but I also think the flow unit (volume per time) are not yet implemented.

**Comment by eike-fokken** [comment]
Oh, another thing ist:

Please make all these maps in the header extern and define them in the cpp file.

**Comment by eike-fokken** [comment]
And reinstate the
```
  extern template
  double parse_to_si(json const &unit_json, std::map<std::string, Conversion> const &unit_map);


  extern template double
  parse_to_si(json const &unit_json,
              std::map<std::string, double> const &unit_map);
```

in the header file. This prevents the compiler from always re-instating the template, when we really only need it to be compiled once.

File: src/Auxillary/unit_conversion/include/unit_conversion.hpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    json const &unit_json,
+    std::map<std::string, conversion> const &unit_map
+  );
+
+
+  extern template 
+  double parse_to_si<double>(
+    json const &unit_json,
+    std::map<std::string, double> const &unit_map
+  );
```
Comment (FelixBenning) [inline src/Auxillary/unit_conversion/include/unit_conversion.hpp:85]: > 
> 
> And reinstate the
> 
> ```
>   extern template
>   double parse_to_si(json const &unit_json, std::map<std::string, Conversion> const &unit_map);
> 
> 
>   extern template double
>   parse_to_si(json const &unit_json,
>               std::map<std::string, double> const &unit_map);
> ```
> 
> in the header file. This prevents the compiler from always re-instating the template, when we really only need it to be compiled once.

they are still there

**Comment by eike-fokken** [comment]
Oh, now I see.
But then please remove the template declaration above them and move the extern template ...

to below your template declaration.

**Comment by FelixBenning** [comment]
- make all these maps in the header extern and define them in the cpp file. https://github.com/eike-fokken/grazer/pull/36/commits/1e2378797720879f642617920c266ec8a53f11d9
- make some more test cases, namely for these units (these all appear in the topology.json) https://github.com/eike-fokken/grazer/pull/36/commits/b57ff883cf450a559eae002cbbdb0e233e0d4335
- remove the template declaration above them and move the extern template https://github.com/eike-fokken/grazer/pull/36/commits/1950937be284e4ca9430890e8def8d0200e46041

**Pull request merged**